### DISCUSSION TITLE
PostSettingsPrePersistHooks

### DISCUSSION THREAD

**Discussion by Machine-Maker** [ory/kratos#2536] [Q&A] [answered]
How can I add a web hook to `PostSettingsPrePersistHooks`? I see it running in the log, but when I add a hook to the after settings flow, it runs in the `PostSettingsPostPersistHooks` hook.

**Comment by vinckr** [comment]
Hello @Machine-Maker,
can you share your configuration including the hook config please? 
I haven't used the pre/post hooks myself yet so need to test it out.

**Reply by Machine-Maker** [reply]
https://github.com/HangarMC/HangarAuth/blob/master/docker/kratos/kratos.yml#L52 here it is. The docs, while mentioning that these pre/post persist hooks exist for things, don't really go into detail at all about how to hook into them.

**Comment by lus** [comment, marked as answer]
I just came across the same issue and after digging through some code I found a solution (that you apparently already implemented in Hangar).
Instead of adding the hook configuration to the `selfservice.flows.settings.after.hooks` key like you would do if you followed the documentation, the strategy has to be included in this key (`selfservice.flows.settings.after.<strategy>.hooks`).
Even though the *possibility* of this additional key is documented, the *requirement* of it in this particular case is not as far as I can see.
For profile updates this would be `profile`, so the final key would be `selfservice.flows.settings.after.profile.hooks`.
I hope I prevent someone stumbling across this issue from spending hours to find a solution.

**Reply by vinckr** [reply]
Thanks for the report lus!
We should think of something to make it clearer you need to use the strategy also in settings hooks in [the documentation](https://www.ory.sh/docs/kratos/hooks/configure-hooks).
